src/tribler-core/tribler_core/containers.py

```diff
@@ -0,0 +1,36 @@
+from ipv8.messaging.interfaces.dispatcher.endpoint import DispatcherEndpoint
+from ipv8.peer import Peer
+
+from ipv8_service import IPv8
+
+from dependency_injector import containers, providers
+
+from tribler_core.ipv8_config import Ipv8Config
+from tribler_core.trustchain_keys import TrustChainKeys
+
+
+class Ipv8Container(containers.DeclarativeContainer):
+    state_dir = providers.Dependency()
+    config = providers.Configuration()
+
+    ipv8_config = providers.Singleton(Ipv8Config, state_dir=state_dir, config=config.provider)
+
+    endpoint = providers.Singleton(
+        DispatcherEndpoint,
+        providers.List(providers.Object("UDPIPv4")),
+        UDPIPv4=providers.Dict(port=providers.Factory(config.port), ip=providers.Factory(config.address)),
```

Comment:
Yep, this line looks not too intuitive :confused:. I would expect it to be just `UDPIPv4={'port':config.port}`, etc. Why the `Factory` stuff is necessary here, what's the semantic reason?

Reply:
It can be replaced by (explanations later):

```python
    endpoint = providers.Singleton(
        DispatcherEndpoint, ["UDPIPv4"],
        UDPIPv4=providers.Dict(port=config.port, ip=config.address),
    )
```

Reply:
If you use `UDPIPv4={'port':config.port}` you'll get:

```python
{'UDPIPv4': {'port': <dependency_injector.providers.ConfigurationOption('config.port') at 0x112961eb0>}}
```